x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -128,33 +147,53 @@ export class TiledVectorLayer extends VectorLayer {
     if (!sourceDataRequest) {
       return;
     }
-    const sourceMeta: MVTSingleLayerVectorSourceConfig = sourceDataRequest.getData() as MVTSingleLayerVectorSourceConfig;
+    const sourceMeta: TiledSingleLayerVectorSourceDescriptor = sourceDataRequest.getData() as MVTSingleLayerVectorSourceConfig;
+
+    if (sourceMeta.layerName === '') {
+      return;
+    }
 
     this._setMbPointsProperties(mbMap, sourceMeta.layerName);
     this._setMbLinePolygonProperties(mbMap, sourceMeta.layerName);
   }
 
   _requiresPrevSourceCleanup(mbMap: unknown): boolean {
-    // @ts-ignore
-    const mbTileSource = mbMap.getSource(this.getId());
+    // @ts-expect-error
+    const mbTileSource = mbMap.getSource(this._getMbSourceId());
     if (!mbTileSource) {
       return false;
     }
+
     const dataRequest = this.getSourceDataRequest();
     if (!dataRequest) {
       return false;
     }
-    const tiledSourceMeta: MVTSingleLayerVectorSourceConfig | null = dataRequest.getData() as MVTSingleLayerVectorSourceConfig;
-    if (
-      mbTileSource.tiles[0] === tiledSourceMeta.urlTemplate &&
-      mbTileSource.minzoom === tiledSourceMeta.minSourceZoom &&
-      mbTileSource.maxzoom === tiledSourceMeta.maxSourceZoom
-    ) {
-      // TileURL and zoom-range captures all the state. If this does not change, no updates are required.
+    const tiledSourceMeta: TiledSingleLayerVectorSourceDescriptor | null = dataRequest.getData() as MVTSingleLayerVectorSourceConfig;
+
+    if (!tiledSourceMeta) {
       return false;
     }
 
-    return true;
+    // check if layername has changed
+    const isSourceDifferent =
+      mbTileSource.tiles[0] !== tiledSourceMeta.urlTemplate ||
+      mbTileSource.minzoom !== tiledSourceMeta.minSourceZoom ||
+      mbTileSource.maxzoom !== tiledSourceMeta.maxSourceZoom;
+
+    if (isSourceDifferent) {
+      return true;
+    }
+
+    const layerIds = this.getMbLayerIds();
+    // just check the first layer
+    for (let i = 0; i < layerIds.length; i++) {
```

Comment:
What is this checking?

Reply:
Each VectorLayer adds about 8 different mapbox layers (fill_*, line_*, icon_*, ...) all pointing at the same mb-source. 

This check will short-circuit if it detects one of those mb-layers where the source-pointer (in this case mbSource#sourceLayer) has gone stale. If one of those mb-layers is stale, they will all be stale, and all need to be removed.

Reply:
It's a little edge-case, due to how mapbox-gl spec works. If `layerName` changes (but all other settings remain identical), the corresponding change in the mapbox-style should _not_ be made at the mb-source, but only for the mb-layers. If the zoom and/or url change, the change needs to be made at the mb-source and mb-layer level.